uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1068,17 +1079,105 @@ impl APIConverter<Field> for weedle::dictionary::DictionaryMember<'_> {
 pub enum Literal {
     Boolean(bool),
     String(String),
-    // TODO: more types of literal
+    // Integers are represented as the widest representation we can.
+    // Number formatting vary with language and radix, so we avoid a lot of parsing and
+    // formatting duplication by using only signed and unsigned variants.
+    UInt(u64, Radix, Type),
+    Int(i64, Radix, Type),
+    // Pass the string representation through as typed in the IDL.
+    // This avoids a lot of uncertainty around precision and accuracy,
+    // though bindings for languages less sophisticated number parsing than WebIDL
+    // will have to do extra work.
+    Float(String, Type),
+    Enum(String, Type),
+    EmptySequence,
+    EmptyMap,
+    Null,
+}
+
+// Represent the radix of integer literal values.
+// We preserve the radix into the generated bindings for readability reasons.
+#[derive(Debug, Clone, Copy, Hash)]
+pub enum Radix {
+    Decimal = 10,
+    Octal = 8,
+    Hexadecimal = 16,
 }
 
-impl APIConverter<Literal> for weedle::literal::DefaultValue<'_> {
-    fn convert(&self, _ci: &mut ComponentInterface) -> Result<Literal> {
-        Ok(match self {
-            weedle::literal::DefaultValue::Boolean(b) => Literal::Boolean(b.0),
-            weedle::literal::DefaultValue::String(s) => Literal::String(s.0.to_string()),
-            _ => bail!("no support for {:?} literal yet", self),
+fn convert_default_value(
+    default_value: &weedle::literal::DefaultValue<'_>,
+    type_: &Type,
+) -> Result<Literal> {
+    fn convert_integer(literal: &weedle::literal::IntegerLit<'_>, type_: &Type) -> Result<Literal> {
+        use regex::Regex;
+
+        let (string, radix) = match literal {
+            weedle::literal::IntegerLit::Dec(v) => (v.0, Radix::Decimal),
+            weedle::literal::IntegerLit::Hex(v) => (v.0, Radix::Hexadecimal),
+            weedle::literal::IntegerLit::Oct(v) => (v.0, Radix::Octal),
+        };
+        // This is the radix of the parsed number, passed to `from_str_radix`.
+        let src_radix = radix as u32;
+        // This radix tells the backends how to represent the number in the output languages.
+        let dest_radix = if string == "0" || string.starts_with('-') {
+            // 1. weedle parses "0" as an octal literal, but we most likely want to treat this as a decimal.
+            // 2. Explicitly negatively signed hex numbers won't convert via i64 very well if they're not 64 bit.
+            //    For ease of implementation, output will use decimal.
+            Radix::Decimal
+        } else {
+            radix
+        };
+
+        let trimmer = Regex::new("^(-?)(0x)?").unwrap();
```

Comment:
I'm not too worried about this, since we're still prototyping and might eventually use more complex regexes elsewhere—but bringing in `regex` can bump up compile times. For prefix-stripping like this, I wonder if we can use plain-ol' string functions, maybe something like this:

```rust
let string = if string.starts_with('-') {
    ("-".to_string() + string[1..].strip_prefix("0x").unwrap_or(&string[1..])).to_lowercase()
} else {
    string.strip_prefix("0x").unwrap_or(&string).to_lowercase()
};
```

WDYT?

Reply:
Oh, magic.

Not since Java 1.3 have I had to think about not having regular expressions.